doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,289 @@ Reasoning with inductive types
-      to introduce. More generally, the :n:`pij` can be any introduction
-      pattern (see :tacn:`intros`). This provides a concise notation for
-      chaining destruction of a hypothesis.
+   .. prodn::
+      induction_clause ::= @induction_arg {? as @or_and_intropattern } {? eqn : @naming_intropattern } {? @occurrences }
+      induction_arg ::= @natural
+      | @one_term_with_bindings
+      induction_principle ::= using @one_term {? with @bindings } {? @occurrences }
 
-   .. tacv:: destruct @term eqn:@naming_intropattern
-      :name: destruct … eqn:
+   Performs case analysis by generating a subgoal for each constructor of the
+   inductive or co-inductive type in the :n:`@one_term` in :n:`@induction_arg`.
+   In each new subgoal, the tactic replaces :n:`@one_term` with the associated
+   constructor (applied to its arguments).   Unlike :tacn:`induction`,
+   :tacn:`destruct` generates no induction hypothesis.
 
-      This behaves as :n:`destruct @term` but adds an equation
-      between :token:`term` and the value that it takes in each of the
-      possible cases. The name of the equation is specified
-      by :token:`naming_intropattern` (see :tacn:`intros`),
-      in particular ``?`` can be used to let Coq generate a fresh name.
+   The :n:`@one_term` may contain holes that are denoted by “_”. In this case,
+   the tactic selects the first subterm that matches the pattern and performs
+   case analysis using that subterm.
 
-   .. tacv:: destruct @term with @bindings
+   :n:`{+, @induction_clause }`
+     Giving multiple :n:`@induction_clause`\s is equivalent to applying :n:`destruct`
+     serially on each :n:`@induction_clause`.
 
-      This behaves like :n:`destruct @term` providing explicit instances for
-      the dependent premises of the type of :token:`term`.
+   :n:`@natural`
+     Equivalent to :tacn:`intros` :n:`until @natural` followed by `destruct` applied to
+     the last introduced hypothesis.
 
-   .. tacv:: edestruct @term
-      :name: edestruct
+     .. note::
+        To destruct a number, use the syntax :n:`destruct (@natural)` (not
+        very interesting anyway).
 
-      This tactic behaves like :n:`destruct @term` except that it does not
-      fail if the instance of a dependent premises of the type
-      of :token:`term` is not inferable. Instead, the unresolved instances
+   :n:`@ident` (in :n:`@induction_arg`)
+     If :token:`ident` is a quantified variable in the conclusion
+     of the goal, then :n:`destruct @ident` behaves like :tacn:`intros`
+     :n:`until @ident; destruct @ident`.
+     If :token:`ident` is a hypothesis of the context and :token:`ident`
+     is no longer dependent in the goal after application
+     of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
+     in :n:`destruct (@ident)`).
+
+   :n:`as @or_and_intropattern`
+      Provides names for the variables introduced in each new subgoal.  The
```

Comment:
It could be clarified that the inner intropatterns are not restricted to naming intropatterns.

Reply:
Is adding this at the end sufficient?

The inner :n:`intropattern`s can also split introduced hypotheses into
multiple hypotheses or subgoals.

Reply:
This goes beyond that: they can also be used to do rewriting (`->` and `<-`), we can apply lemmas to the introduced hypotheses `%` or injection (`[= ]`)... Basically, the full power of intropatterns is available for inner intropatterns (but only for inner intropatterns).

Reply:
Is it really necessary to repeat everything that `intropattern` can do given that text already hyperlinks to the `intropattern` nonterminal?  And if so, what wording do you suggest?

Reply:
No, it would be a terrible idea to repeat everything that intro patterns can do because it would quickly become outdated. I just suggest to clarify that not just naming patterns are supported but the whole set (as long as they are nested).

Reply:
Then how about:

> Note the inner :n:`intropattern`s are not restricted to renaming; they can be used to split introduced hypotheses into multiple hypotheses, to do rewriting (`->` and `<-`), to apply lemmas to the introduced hypotheses (`%`) or to do `injection` (`[= ... ]`).

I really don't understand why you want this tactic to have this extra detail that's covered in in the intropattern section.  If you don't like this wording, please provide specific wording.

Reply:
What you propose is fine.

>I really don't understand why you want this tactic to have this extra detail that's covered in in the intropattern section.

**I don't!** I just want to clarify that intropatterns are not semantically restricted to naming. This is what is suggested by the current sentence "Provides names for the variables introduced in each new subgoal."

The change could have been as simple as:

>Provide names for (or apply further transformations to) the introduced variables and hypotheses in each subgoal.

Reply:
Got it.  It is now:

Provides names for (or apply further transformations to)
the variables and hypotheses introduced in each new subgoal.